flow/workflows/cdc_flow.go

```diff
@@ -91,69 +91,104 @@ func processCDCFlowConfigUpdate(
 ) error {
 	flowConfigUpdate := state.FlowConfigUpdate
 
-	if flowConfigUpdate != nil {
-		logger.Info("processing CDCFlowConfigUpdate", slog.Any("updatedState", flowConfigUpdate))
-		if len(flowConfigUpdate.AdditionalTables) == 0 {
-			return nil
-		}
-		if shared.AdditionalTablesHasOverlap(state.SyncFlowOptions.TableMappings, flowConfigUpdate.AdditionalTables) {
-			logger.Warn("duplicate source/destination tables found in additionalTables")
-			return nil
-		}
-		state.CurrentFlowStatus = protos.FlowStatus_STATUS_SNAPSHOT
+	defer func() {
+		syncStateToConfigProtoInCatalog(ctx, logger, cfg, state)
```

Comment:
seems risky, should probably skip this if handling panic

Reply:
added manually to all good return spots